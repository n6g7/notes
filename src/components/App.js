import React, { PureComponent } from "react"
import { Switch, Route } from "react-router-dom"

import { Notebook, Summary } from "@pages"

class App extends PureComponent {
  render() {
    return (
      <Switch>
        <Route exact path="/" component={Summary} />
        <Route path="/nb/:notebook" component={Notebook} />
      </Switch>
    )
  }
}

export default App
