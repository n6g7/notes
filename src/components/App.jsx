import React, { PureComponent } from 'react';
import { Link } from 'react-router';

import githubIcon from '../images/github.svg';
import notebooks from '../../notebooks';

class App extends PureComponent
{
  render() {
    return <main>
      <header>
        <h1>notes</h1>
        <nav>
          <ul>
            { Object.keys(notebooks).map(name =>
              <li key={`nb-${name}`}>
                <Link to={`/nb/${name}`}>{name}</Link>
              </li>
            )}
          </ul>
          <aside>
            <a href="https://github.com/n6g7/notes" title="GitHub repository" target="blank">
              <img src={githubIcon} alt="GitHub" />
            </a>
          </aside>
        </nav>
      </header>
      <article className="container">
        {this.props.children}
      </article>
    </main>;
  }
}

export default App;
