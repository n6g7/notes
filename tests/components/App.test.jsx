import React from 'react'
import App from '../../src/components/App.jsx'
import renderer from 'react-test-renderer'

describe('App', () => {
  it('renders a well formed component', () => {
    const component = renderer.create(<App />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
