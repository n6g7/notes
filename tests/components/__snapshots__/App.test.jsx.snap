exports[`App renders a well formed component 1`] = `
<main>
  <header>
    <h1>
      <a
        onClick={[Function]}
        style={Object {}}>
        notes
      </a>
    </h1>
    <nav>
      <ul>
        <li>
          <a
            onClick={[Function]}
            style={Object {}}>
            bertrand
          </a>
        </li>
        <li>
          <a
            onClick={[Function]}
            style={Object {}}>
            secretary
          </a>
        </li>
        <li>
          <a
            onClick={[Function]}
            style={Object {}}>
            simulation
          </a>
        </li>
      </ul>
      <aside>
        <a
          href="https://github.com/n6g7/notes"
          target="blank"
          title="GitHub repository">
          <img
            alt="GitHub"
            src={Object {}} />
        </a>
      </aside>
    </nav>
  </header>
  <article
    className="container" />
</main>
`;
